src/sys/reboot.rs

```diff
@@ -0,0 +1,44 @@
+use {Errno, Error, Result};
+use libc;
+use void::Void;
+use std::mem::drop;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum RebootMode {
+    Halt,
+    kexec,
+    PowerOff,
+    Restart,
+    // we do not support Restart2,
+    Suspend,
+}
+
+pub fn reboot(how: RebootMode) -> Result<Void> {
+    let cmd = match how {
+        RebootMode::Halt => libc::RB_HALT_SYSTEM,
+        RebootMode::kexec => libc::RB_KEXEC,
+        RebootMode::PowerOff => libc::RB_POWER_OFF,
+        RebootMode::Restart => libc::RB_AUTOBOOT,
+        // we do not support Restart2,
+        RebootMode::Suspend => libc::RB_SW_SUSPEND,
+    };
+    unsafe {
+        libc::reboot(cmd)
+    };
+    Err(Error::Sys(Errno::last()))
+}
+
+/// Enable or disable the reboot keystroke (Ctrl-Alt-Delete).
+///
+/// Corresponds to calling `reboot(RB_ENABLE_CAD)` or `reboot(RB_DISABLE_CAD)` in C.
```

Comment:
Could we mention in `RebootMode`'s documentation, that this function exists, in case someone is looking for that functionality.